scripts/irods/test/test_icommands_file_operations.py

```diff
@@ -1631,3 +1631,64 @@ def test_writing_collection_under_slash_4030(self):
             self.user0.run_icommand('irm -rf {dir1}'.format(**locals()))
             shutil.rmtree(os.path.abspath(dir1path), ignore_errors=True)
 
+    # These tests create a resource with a vault for which iRODS has no write permission and tries to put a file there
+    def test_iput_small_file_to_resource_with_restricted_vault_permission(self):
+        self.iput_to_resource_with_restricted_vault_permission_test(1)
+
+    def test_iput_large_file_to_resource_with_restricted_vault_permission(self):
+        self.iput_to_resource_with_restricted_vault_permission_test(40000001)
+
+    def iput_to_resource_with_restricted_vault_permission_test(self, size):
+        resc_name = 'cantwritetovaultresc'
+        vault_path = os.path.join('/', 'var')
+        self.admin.assert_icommand(['iadmin', 'mkresc', resc_name, 'unixfilesystem', lib.get_hostname() + ':' + vault_path], 'STDOUT_SINGLELINE', resc_name)
+        file_name = 'test_iput_to_resource_with_restricted_vault_permission'
+        file_path = os.path.join(self.testing_tmp_dir, file_name)
+        lib.make_file(file_path, size)
+        logical_path = os.path.join(self.admin.session_collection, file_name) # another user's home collection
+        try:
+            self.admin.assert_icommand(['iput', '-R', resc_name, file_path], 'STDERR', 'UNIX_FILE_MKDIR_ERR')
+            self.admin.assert_icommand(['ils', '-l', file_name], 'STDERR', 'does not exist')
+            session_vault_path = self.admin.get_vault_session_path()
+            self.assertTrue(False == os.path.exists(os.path.join(session_vault_path, file_name)))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [comparison to False should be 'if cond is False:' or 'if not cond:'](https://app.codacy.com/manual/trel/irods/pullRequest?prid=4331469)